typeclasses/npcshop/npcshop.py

```diff
@@ -257,13 +258,16 @@ def at_object_creation(self):
         self.cmdset.add_default(StoreRoomCmdSet)
 
     def return_appearance(self, caller):
-        wares = []
-        for item in self.contents:
-            if item.db.value:
-                wares.append(item.key)
+        wares = [item for item in self.contents if item.db.value]
         desc = super(StoreRoom, self).return_appearance(caller)
         for line in desc.split("\n"):
-            if line.strip().split("(")[0] in wares:
-                line = "|y" + line
+            if any(line.find(ware.key) > -1 for ware in wares):
+                try:
+                    line = "|y{line} |n({price})".format(
```

Comment:
So I wrote the following last night but forgot to submit it before your comment this morning:
-=-=-

Given the example in your gist, then we really are talking about fully re-implementing `return_appearance` as far as Ainneve rooms are concerned, but with more extensible ways of organizing the data. I can get behind that idea, but it's starting to range beyond the scope of creating a usable NPC shop. 

Since Ainneve is using `contrib.rpsystem` components, maybe the right way to move forward for now, is to grab the guts of `ContribRPObject.return_appearance` and modify it within the `StoreRoom` class to get it working and finish off this PR. 

Then, create a new issue/branch where you can address re-organizing how the base `Room` typeclass marshals its data and make it easier for subclasses to reformat their own `return_appearance` output.